DNN Platform/Library/Common/Utilities/HtmlUtils.cs

```diff
@@ -521,28 +521,36 @@ public static void WriteFeedback(HttpResponse response, Int32 indent, string mes
         /// -----------------------------------------------------------------------------
         public static void WriteFeedback(HttpResponse response, Int32 indent, string message, bool showtime)
         {
-            bool showInstallationMessages = true;
-            string ConfigSetting = Config.GetSetting("ShowInstallationMessages");
-            if (ConfigSetting != null)
-            {
-                showInstallationMessages = bool.Parse(ConfigSetting);
-            }
-            if (showInstallationMessages)
+            try
             {
-                //Get the time of the feedback
-                TimeSpan timeElapsed = Upgrade.RunTime;
-                string strMessage = "";
-                if (showtime)
+                bool showInstallationMessages = true;
+                string ConfigSetting = Config.GetSetting("ShowInstallationMessages");
+                if (ConfigSetting != null)
                 {
-                    strMessage += timeElapsed.ToString().Substring(0, timeElapsed.ToString().LastIndexOf(".", StringComparison.Ordinal) + 4) + " -";
+                    showInstallationMessages = bool.Parse(ConfigSetting);
                 }
-                for (int i = 0; i <= indent; i++)
+                if (showInstallationMessages)
                 {
-                    strMessage += "&nbsp;";
+                    //Get the time of the feedback
+                    TimeSpan timeElapsed = Upgrade.RunTime;
+                    string strMessage = "";
+                    if (showtime)
+                    {
+                        strMessage += timeElapsed.ToString().Substring(0, timeElapsed.ToString().LastIndexOf(".", StringComparison.Ordinal) + 4)
+                                      + " -";
+                    }
+                    for (int i = 0; i <= indent; i++)
+                    {
+                        strMessage += "&nbsp;";
+                    }
+                    strMessage += message;
+                    response.Write(strMessage);
+                    response.Flush();
                 }
-                strMessage += message;
-                response.Write(strMessage);
-                response.Flush();
+            }
+            catch (HttpException)
```

Comment:
Better to avoid exceptions in the first place, see comment on left

Reply:
The problem is no tin the parsing; it is in writing to a closed stream; therefore, we still need the try/catch blocks.